lib/metadata/linux/LinuxUtils.rb

```diff
@@ -193,5 +214,52 @@ def self.parse_openstack_status(lines)
         }
       end
     end
+
+    def self.parse_openstack_container_status(lines)
+      lines.to_s.split("\n").group_by do |t|
+        if t.include?('_')
+          t.split("_")[0]
+        elsif t.start_with?('openstack')
+          t.split("-")[1]
+        elsif t.include?('-')
+          t.split("-")[0]
+        else
+          t.split(" ")[0]
+        end
+      end.map do |title, services|
+        {
+          'name'     => title.capitalize,
+          'services' => services.map do |service_line|
+            # split service line by first space and strip white space from results
+            service_line.split(/^(\S+)\s/)
+          end.map do |service|
+            {
+              'name'    => service[1],
+              'active'  => service[2].strip.start_with?('Up'),
+              'enabled' => true
+            }
+          end
+        }
+      end
+    end
+
+    def self.merge_openstack_services(systemctl_services, containerized_services)
```

Comment:
Probably a good idea to have some explanatory comments in here, or at least an introductory comment that explains what this does in a sentence.